src/libraries/System.IO.FileSystem/tests/FileStream/ctor_str_fm_fa_fs_buffer.cs

```diff
@@ -26,10 +26,11 @@ public void NegativeBufferSizeThrows()
         }
 
         [Fact]
-        public void ZeroBufferSizeThrows()
+        public void ZeroBufferSizeDoesNotThrow()
         {
-            // Unfortunate pre-existing behavior of FileStream, we should look into enabling this sometime.
-            AssertExtensions.Throws<ArgumentOutOfRangeException>("bufferSize", () => CreateFileStream(GetTestFilePath(), FileMode.Create, FileAccess.ReadWrite, FileShare.Read, 0));
+            using (CreateFileStream(GetTestFilePath(), FileMode.Create, FileAccess.ReadWrite, FileShare.Read, 0))
+            {
+            }
         }
```

Comment:
I think it'd be better to just change ValidBufferSize below to be a theory, with `[InlineData(0)]`, `[InlineData(64 * 1024)]`, and whatever other sizes we want to validate.